plugins/ssr/ssripats.ml

```diff
@@ -137,17 +235,68 @@ let tacCHECK_HYPS_EXIST hyps = Goal.enter begin fun gl ->
   tclUNIT ()
 end
 
+let name_seeds_of_ind env sigma ty =
+  let ((kn, i), _ as indu), unfolded_c_ty =
+    Tacred.reduce_to_quantified_ind env sigma ty
+  in
+  let mind, indb = Inductive.lookup_mind_specif env (kn,i) in
+  let params = mind.Declarations.mind_nparams in
+  let drop_params x =
+    snd @@ EConstr.decompose_prod_n_assum sigma params (EConstr.of_constr x) in
+  Array.map drop_params indb.Declarations.mind_nf_lc
+
 (** [=> []] *****************************************************************)
+
+let tclTHENin t1 t2 =
+  tclUNIT () >>= begin fun () -> let i = ref (-1) in
+  t1 <*> numgoals >>= fun n ->
+  Goal.enter begin fun g -> incr i; t2 !i n end
+end
+
+let tclSEED_SUBGOALS seeds tac =
```

Comment:
Can you explain what this does in a comment?

Reply:
done